storage/src/key_value/agent/mod.rs

```diff
@@ -0,0 +1,199 @@
+        trace!("received message: {}", message);
+        match message {
+            Message::InsertBlock(block) => Box::new(self.wrap(move |f| f.insert_block(block))),
+            Message::DeleteBlock(hash) => Box::new(self.wrap(move |f| f.delete_block(&hash))),
+            Message::GetBlockHash(block_num) => Box::new(self.get_block_hash(block_num)),
+            Message::GetBlockHeader(hash) => Box::new(self.get_block_header(&hash)),
+            Message::GetBlockState(hash) => Box::new(self.get_block_state(&hash)),
+            Message::GetBlockStates(hashes) => Box::new(self.get_block_states(&hashes[..])),
+            Message::GetBlock(hash) => Box::new(self.get_block(&hash)),
+            Message::GetForkPath(hash, oldest_fork_threshold) => {
+                Box::new(self.get_fork_path(&hash, oldest_fork_threshold))
+            }
+            Message::CommitBlock(block_hash, ledger_digest) => {
+                Box::new(self.wrap(move |f| f.commit_block(&block_hash, &ledger_digest)))
+            }
+            Message::DecommitBlocks(hash) => Box::new(self.wrap(move |f| f.decommit_blocks(&hash))),
+            Message::Canon() => Box::new(self.canon()),
+            Message::LongestChildPath(hash) => Box::new(self.longest_child_path(&hash)),
+            Message::GetBlockLocatorHashes() => Box::new(self.get_block_locator_hashes()),
+            Message::FindSyncBlocks(hashes, block_count) => Box::new(self.find_sync_blocks(hashes, block_count)),
+            Message::GetTransactionLocation(transaction_id) => Box::new(self.get_transaction_location(&transaction_id)),
+            Message::GetRecordCommitments(limit) => Box::new(self.get_record_commitments(limit)),
+            Message::GetRecord(commitment) => Box::new(self.get_record(&commitment)),
+            Message::StoreRecords(records) => Box::new(self.wrap(move |f| f.store_records(records))),
+            Message::GetCommitments() => Box::new(self.get_digest_keys(KeyValueColumn::Commitment)),
+            Message::GetSerialNumbers() => Box::new(self.get_digest_keys(KeyValueColumn::SerialNumber)),
+            Message::GetMemos() => Box::new(self.get_digest_keys(KeyValueColumn::Memo)),
+            Message::GetLedgerDigests() => Box::new(self.get_ledger_digests()),
+            Message::ResetLedger(commitments, serial_numbers, memos, digests) => {
+                Box::new(self.wrap(move |f| f.reset_ledger(commitments, serial_numbers, memos, digests)))
+            }
+            Message::GetCanonBlocks(limit) => Box::new(self.get_canon_blocks(limit)),
+            Message::GetBlockHashes(limit, filter) => Box::new(self.get_block_hashes(limit, filter)),
+        }
+    }
+
+    pub(super) fn agent(mut self, mut receiver: mpsc::Receiver<MessageWrapper>) {
+        while let Some((message, response)) = receiver.blocking_recv() {
+            let out = self.handle_message(message);
+            response.send(out).ok();
```

Comment:
shouldn't we at least `warn!` when this fails due to the channel being full? this is a bigger deal than a network message not being delivered/received

Reply:
these are oneshot channels, and it being dropped implies the caller didn't care about the result, which is acceptable behavior. rust will provide compiler warnings since they are results.

Reply:
ah, ok, I must have mixed it up with some fixed-capacity `mpsc::Receiver` :+1: